### PR TITLE
Improve selecting projects

### PR DESCRIPTION
**Pull request by Tymek**
Adds tests and improvements to the form for creating API tokens.

### CODE REVIEW

**Comment by Tymek** [comment]
[API token: improve client tokens (additional Unit Test)](https://trello.com/c/e1Q6VIqC/1034-api-token-improve-client-tokens-additional-unit-test)

File: src/component/admin/apiToken/ApiTokenForm/SelectProjectInput/SelectAllButton/SelectAllButton.tsx [hunk 1]
```diff
@@ -15,7 +15,11 @@ export const SelectAllButton: FC<SelectAllButtonProps> = ({
 
     return (
         <Box sx={{ ml: 3.5, my: 0.5 }}>
-            <Link onClick={onClick} className={styles.selectOptionsLink}>
+            <Link
+                onClick={onClick}
+                className={styles.selectOptionsLink}
+                role="button"
```
Comment (olav) [inline src/component/admin/apiToken/ApiTokenForm/SelectProjectInput/SelectAllButton/SelectAllButton.tsx:21]: Looks like this results in an `<a role="button">`. We should instead just use a `<button>`.

File: src/component/admin/apiToken/ApiTokenForm/SelectProjectInput/SelectProjectInput.tsx [hunk 2]
```diff
@@ -79,13 +83,20 @@ export const SelectProjectInput: VFC<ISelectProjectInputProps> = ({
...
-                    setProjects(
-                        isAllSelected ? [] : options.map(({ value }) => value)
-                    );
-                }}
+            <ConditionallyRender
+                condition={options.length > 2}
+                show={
+                    <SelectAllButton
+                        isAllSelected={isAllSelected}
+                        onClick={() => {
```
Comment (olav) [inline src/component/admin/apiToken/ApiTokenForm/SelectProjectInput/SelectProjectInput.tsx:91]: Could extract this into a `const onSelectAllClick = ...`, just for readability.

Comment (Tymek) [reply src/component/admin/apiToken/ApiTokenForm/SelectProjectInput/SelectProjectInput.tsx]: refactored 🙂

**Review by olav — APPROVED** [review]
:tada:

**Pull request merged**